sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionEvaluatesToConstant.cs

```diff
@@ -85,18 +93,17 @@ public sealed class ConditionEvaluatesToConstant : SonarDiagnosticAnalyzer
         };
 
         private const string S2583DiagnosticId = "S2583"; // Bug
-        private const string S2583MessageFormat = "Change this condition so that it does not always evaluate to '{0}'; some subsequent code is never executed.";
+        private const string S2583MessageFormatBool = "Change this condition so that it does not always evaluate to '{0}'; some subsequent code is never executed.";
+        private const string S2583MessageFormatNull = "Change this expression which always evaluates to '{0}'.";
```

Comment:
I'd keep the `; some subsequent code is never executed.` part of the message, as the rule S2583 is really about unreachable code

Reply:
The rule title is `Conditionally executed blocks should be reachable`. Shouldn't we update it to `Conditionally executed code should be reachable`? In order to cover expressions with ?? and ??= as well as no-block single liners:
```
if(false)
  DoSomething();
```
Blocks seems to restrictive, not broad enough. But the rename would affect other languages as well - can we do it?

Reply:
Makes sense to me